tex/generic/pgf/frontendlayer/tikz/libraries/tikzlibrarycalendar.code.tex

```diff
@@ -86,73 +78,82 @@
 \def\tikzdaycode{\node[name=\pgfcalendarsuggestedname,every day]{\tikzdaytext};}
 \def\tikzmonthcode{\node[every month]{\tikzmonthtext};}
 \def\tikzyearcode{\node[every year]{\tikzyeartext};}
+\def\tikzweekcode{\node[every week]{\tikzweektext};}
 
 \def\tikzdaytext{\%d-}
 \def\tikzmonthtext{\%mt}
 \def\tikzyeartext{\%y0}
+\def\tikzweektext{\%n=}
 
+\tikzset{
+  week code/.code=\def\tikzweekcode{#1},
+  week text/.code=\def\tikzweektext{#1},
```

Comment:
```suggestion
  week code/.store in=\tikzweekcode,
  week text/.store in=\tikzweektext,
```

Reply:
Can we do this consistently? Why not change every key to a `.store in` that does only a `\def\macro{#1}`? The library still uses for example
```latex
\tikzoption{day code}{\def\tikzdaycode{#1}}
```
Why not change this to `day code/.store in=\tikzaycode`?

And why not use an actual value key (`day code/.initial=\%d-`) here, too? That way one can use `day code/.append=<stuff>` or similar things. And then just `\def\tikzdaycode{\pgfkeysvalueof{/tikz/day code}}` for backwards compatibility and ease of use.

As I'm sure you aware, there are a lot of either `/.code=\def\macro{#1}` or even `\tikzoption{x}{\def\macro{#1}}` still in TikZ.

Reply:
You're welcome to replace all of them if you write the regression tests.